mobile/views/DirectMessages/components/DirectMessageThreadsList.js

```diff
@@ -31,10 +32,13 @@ const DirectMessageThreadsList = (props: Props) => {
   const { isLoading, hasError, data: { user }, navigation } = props;
   if (user) {
     const { pageInfo, edges } = user.directMessageThreadsConnection;
+    const nodes = edges.map(e => e && e.node);
+    const unique = deduplicateChildren(nodes, 'id');
```

Comment:
Why would the server ever send duplicate DM threads?! :scream: We should fix that server-side!

Reply:
I don't think it's the server, I think it's the infinite scroll component firing a `fetchMore` multiple times while it's still `refetch`ing - might need to just add in some `shouldComponentUpdate` logic to try and prevent this from happening.

Reply:
It might also be the server, too - not sure if we have the same problem in the web app

Reply:
I thought I made the `InfiniteList` be very clear about only ever calling `fetchMore` once?

https://github.com/withspectrum/spectrum/blob/ba93c60928218dc1436a8c754324d012321a3a9f/mobile/components/InfiniteList/index.js#L51-L61

Reply:
Same - I haven't duplicated any of the code, it's just my hunch at what's happening.

Reply:
Again I'm not sure what's going on. It looks like we *do* deduplicate dm threads in the web app as well, so maybe it is a server issue.